src/proof/dot/dot_printer.h

```diff
@@ -19,13 +19,50 @@
 #define CVC5__PROOF__DOT__DOT_PRINTER_H
 
 #include <iostream>
+#include <stack>
 
 #include "printer/let_binding.h"
 #include "proof/proof_node.h"
 
 namespace cvc5::internal {
 namespace proof {
 
+/**
+ * An enumeration for nodes clusterS type. Each type defines in which cluster
```

Comment:
```suggestion
 * An enumeration for nodes cluster types. Each type defines in which cluster
```

Another thing: here and everywhere refer to "proof nodes" rather than "nodes". This is to avoid confusion with between `ProofNode` and `Node`